met-web/src/components/survey/create/CreateSurveyContext.tsx

```diff
@@ -21,7 +37,35 @@ interface SurveyForm {
 }
 
 export const CreateSurveyContextProvider = ({ children }: { children: JSX.Element }) => {
+    const dispatch = useAppDispatch();
+    const navigate = useNavigate();
     const [surveyForm, setSurveyForm] = useState<SurveyForm>(initialSurveyForm);
+    const [loading, setLoading] = useState(true);
+    const [engagementToLink, setEngagementToLink] = useState<Engagement | null>(null);
+    const [availableSurveys, setAvailableSurveys] = useState<Survey[] | null>(null);
+
+    const location = useLocation();
+    const searchParams = new URLSearchParams(location.search);
+    const engagementId = searchParams.get('engagementId');
+
+    useEffect(() => {
+        fetchEngagementToLink();
+    }, [engagementId]);
+
+    const fetchEngagementToLink = async () => {
+        if (!engagementId || isNaN(Number(engagementId))) {
```

Comment:
!engagementId || isNaN(Number(engagementId))
looks like a complicated check... do we need the isNan and all?

Reply:
Just in case someone played with the URL and added '$%^&*' for engagementId